worker/worker-execution/score-worker-execution-impl/src/main/java/io/cloudslang/worker/execution/services/ExecutionServiceImpl.java

```diff
@@ -579,10 +579,12 @@ private void setWorkerGroup(Execution execution) {
             ExecutionStep nextStep = getNextStep(execution);
             if (nextStep != null && nextStep.getActionData().get("workerGroup") != null) {
                 group = nextStep.getActionData().get("workerGroup").toString();
+                execution.setGroupName(group);
```

Comment:
I don't think this is going to work since you are changing the worker group before you persist it.
Concrete example: you have 2 steps, step 1 on wg1, step2 on wg2. Before step1 the group will be wg1, after step 1 group will be wg2, after you do this the information is persisted in the db and then processed in the end step -> group for step1 will be wg2 instead of wg1. Correct me if I am wrong.

Reply:
only that this action is performed after the event was triggered, in the postExecutionSettings
It works, I've tested it